app/controllers/ReportingController.scala

```diff
@@ -83,6 +93,38 @@ trait ReportingController extends BaseController {
     }
   }
 
+
+  def onlineTestPassMarkReport(frameworkId: String) = Action.async { implicit request =>
+    val reports =
+      for {
+        applications <- appRepository.onlineTestPassMarkReport(frameworkId)
+        questionnaires <- questionnaireRepository.findForOnlineTestPassMarkReport
+      } yield {
+        for {
+          a <- applications
+          q <- questionnaires.get(a.applicationId)
+        } yield OnlineTestPassMarkReportItem(a, q)
+      }
+
+    reports.map { list =>
+      Ok(Json.toJson(list))
+    }
+  }
+
+
+  def nonSubmittedAppsReport(frameworkId: String) =
+    preferencesAndContactReports(nonSubmittedOnly = true)(frameworkId)
+
+  def createPreferencesAndContactReports(frameworkId: String) =
+    preferencesAndContactReports(nonSubmittedOnly = false)(frameworkId)
+
+  def applicationAndUserIdsReport(frameworkId: String) = Action.async { implicit request =>
+    appRepository.allApplicationAndUserIds(frameworkId).map { list =>
+      Ok(Json.toJson(list))
+    }
+  }
+
+  /*
```

Comment:
I decide to comment what it is not used right now because it was very confusing for me a file that long, commenting it helped me alot. I am a bit scared of removing it completely, although I should. I check with BAs and they are not sure what report we will have in the future.